src/inputs/ObjectSearch.tsx

```diff
@@ -72,6 +76,28 @@ class ObjectSearch extends Component<IObjectSearchProps> {
     return !!this.options.length;
   }
 
+  private updateValueCaches () {
+    this.previousEndpoint = this.fieldConfig.endpoint;
+    this.previousSearchFilters = toKey(this.fieldConfig.searchFilters || {});
+  }
+
+  private get hasNewEndpoint () {
+    return this.previousEndpoint === this.fieldConfig.endpoint;
+  }
+
+  private get hasNewSearchFilters () {
+    return this.previousSearchFilters === toKey(this.fieldConfig.searchFilters || {});
+  }
+
+  private get isPristine () {
+    return (
+      !this.hasOptions
+      && !this.hasSearch
+      && !this.hasNewEndpoint
```

Comment:
I believe this boolean logic is incorrect. You do want to search if there is a new endpoint or new search filters